test/hotspot/gtest/runtime/test_os.cpp

```diff
@@ -1113,3 +1113,72 @@ TEST_VM(os, free_without_uncommit) {
   os::release_memory(base, size);
 }
 #endif
+
+TEST_VM(os, commit_memory_or_exit) {
+  const size_t page_sz = os::vm_page_size();
+  const size_t size = 16 * page_sz;
+  const char* letters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+  char* base = os::reserve_memory(size, false, mtTest);
+  ASSERT_NE(base, (char*) nullptr);
+  os::commit_memory_or_exit(base, size, false, "Commit failed.");
+  strcpy(base, letters);
+  ASSERT_TRUE(os::uncommit_memory(base, size, false));
+  os::commit_memory_or_exit(base, size, page_sz, false, "Commit with alignment hint failed.");
+  strcpy(base, letters);
+  ASSERT_TRUE(os::uncommit_memory(base, size, false));
+  EXPECT_TRUE(os::release_memory(base, size));
+}
+
+#if !defined(_AIX)
+TEST_VM(os, map_memory_to_file) {
+  const char* letters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  const size_t size = strlen(letters) +1;
+
+  int fd = os::open("map_memory_to_file.txt", O_RDWR | O_CREAT, 0666);
+  EXPECT_TRUE(fd > 0);
+  EXPECT_TRUE(os::write(fd, letters, size));
+
+  char* result = os::map_memory_to_file(size, fd, mtTest);
+  EXPECT_TRUE(result != nullptr);
+
+  EXPECT_TRUE(strcmp(letters, result)==0);
```

Comment:
```suggestion
  EXPECT_TRUE(strcmp(letters, result) == 0);
```
but probably even better:
```suggestion
  EXPECT_EQ(strcmp(letters, result), 0);
```

Reply:
There are more places like this.